src/friendly-iframe-embed.js

```diff
@@ -0,0 +1,139 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from './log';
+import {extensionsFor} from './extensions';
+import {loadPromise} from './event-helper';
+import {getTopWindow, setParentWindow} from './service';
+
+
+/**
+ * Parameters used to create the new "friendly iframe" embed.
+ * - html: The complete content of an AMP embed, which is itself an AMP
+ *   document. Can include whatever is normally allowed in an AMP document,
+ *   except for AMP `<script>` declarations. Those should be passed as an
+ *   array of `extensionIds`.
+ * - extensionsIds: An optional array of AMP extension IDs used in this embed.
+ * - fonts: An optional array of fonts used in this embed.
+ *
+ * @typedef {{
+ *   html: string,
+ *   extensionIds: (?Array<string>|undefined),
+ *   fonts: (?Array<string>|undefined),
+ * }}
+ */
+export let FriendlyIframeSpec;
+
+
+/**
+ * @type {boolean|undefined}
+ * @visiblefortesting
+ */
+let friendlyIframeSupported;
+
+/**
+ * @param {boolean|undefined} val
+ * @visiblefortesting
+ */
+export function setFriendlyIframeSupportedForTesting(val) {
+  friendlyIframeSupported = val;
+}
+
+/**
+ * Returns `true` if the Friendly Iframes are supported.
+ * @return {boolean}
+ */
+export function isFriendlyIframeSupported() {
+  if (friendlyIframeSupported === undefined) {
+    friendlyIframeSupported = 'srcdoc' in HTMLIFrameElement.prototype;
+  }
+  return friendlyIframeSupported;
+}
+
+
+/**
+ * Creates the requested "friendly iframe" embed. Returns the promise that
+ * will be resolved as soon as the embed is available. The actual
+ * initialization of the embed will start as soon as the `iframe` is added
+ * to the DOM.
+ * @param {!HTMLIFrameElement} iframe
+ * @param {!FriendlyIframeSpec} spec
+ * @return {!Promise<FriendlyIframeEmbed>}
+ */
+export function createFriendlyIframeEmbed(iframe, spec) {
+  dev().assert(isFriendlyIframeSupported(), 'friendly iframes not supported');
+
+  const win = getTopWindow(iframe.ownerDocument.defaultView);
+  const extensions = extensionsFor(win);
+
+  iframe.style.visibility = 'hidden';
+  // DO NOT SUBMIT: REVIEW: What's the best value for referrer policy here?
+  iframe.setAttribute('referrerpolicy', 'unsafe-url');
+
+  // Pre-load extensions.
+  if (spec.extensionIds) {
+    spec.extensionIds.forEach(
+        extensionId => extensions.loadExtension(extensionId));
+  }
+
+  // Load fonts.
+  if (spec.fonts) {
+    // DO NOT SUBMIT: REVIEW: Fonts can be shared via top doc. Is this a good idea?
```

Comment:
Here I'm installing them in the parent doc as with SD. So, it sounds like you'd rather only have them in IFRAME?